test/Microsoft.SqlTools.ServiceLayer.UnitTests/LanguageServer/LanguageServiceTests.cs

```diff
@@ -210,5 +214,45 @@ public void GetDefaultCompletionListWithMatchesTest()
             Assert.AreEqual(1, result.Length);
 
         }
+
+        [Test]
+        public async Task CheckForNonTSqlLanguageTest()
+        {
+            // get the test service 
+            LanguageService service = TestObjects.GetTestLanguageService();
+            var connInfo = new ConnectionInfo(null, null, null);
+
+            const string tsqlText = "SELECT * FROM sys.objects\n--returning\nselect * s\n";
+            const string nonTSqlText = "returning";
+            string errorLimitText = string.Concat(Enumerable.Repeat("select * s\n", 51));
+
+            var scriptFileTSql = new ScriptFile();
+            scriptFileTSql.SetFileContents(tsqlText);
+
+            var scriptFileNonTSql = new ScriptFile();
+            scriptFileNonTSql.SetFileContents(nonTSqlText);
+
+            var errorScriptFile = new ScriptFile();
+            errorScriptFile.SetFileContents(errorLimitText);
+
+            ParseResult tSqlParseResult = service.ParseAndBind(scriptFileTSql, connInfo).GetAwaiter().GetResult();
+            ParseResult nonTSqlParseResult = service.ParseAndBind(scriptFileNonTSql, connInfo).GetAwaiter().GetResult();
+            ParseResult errorParseResult = service.ParseAndBind(errorScriptFile, connInfo).GetAwaiter().GetResult();
+            
+            // This should not notify, so it won't error out
+            await service.CheckForNonTSqlLanguage(scriptFileTSql.ClientUri, tSqlParseResult);
+
+            // Check that send event is called; this will error out because the Language Service's 
+            // service host is not set up; it will only error out if a Non T Sql Notification is trying to be sent out
+            var exceptionErrorFile = Assert.ThrowsAsync<System.InvalidOperationException>(async () =>
+               await service.CheckForNonTSqlLanguage(errorScriptFile.ClientUri, errorParseResult));
+            Assert.AreEqual("SendEvent called when ProtocolChannel was not yet connected", exceptionErrorFile.Message);
```

Comment:
What are we testing here exactly? Shouldn't we just assert true/false? You can try isolating your core logic to bare minimum so that it is easier to test.

Reply:
We shouldn't be testing indirect events that might occur because of certain things in our core logic. 

It will be better if we are just testing true/false for these tests.